clients/mercury.go

```diff
@@ -44,11 +44,25 @@ func (client *mercuryClient) NetworkParams() *chaincfg.Params {
 
 func (client *mercuryClient) GetUTXOs(ctx context.Context, address string, limit, confitmations int64) ([]UTXO, error) {
 	utxos := []UTXO{}
-	resp, err := http.Get(fmt.Sprintf("%s/utxo/%s?limit=%d&confirmations=%d", client.URL, address, limit, confitmations))
-	if err != nil || resp.StatusCode != http.StatusOK {
+	var resp *http.Response
+	var err error
+	for {
+		resp, err = http.Get(fmt.Sprintf("%s/utxo/%s?limit=%d&confirmations=%d", client.URL, address, limit, confitmations))
+		if err == nil && resp.StatusCode == http.StatusOK {
+			break
+		}
+
 		if err != nil {
 			return utxos, err
 		}
+
+		// In this case something bad has happened and we should try the
+		// request again
+		if resp.StatusCode >= 500 {
+			fmt.Printf("bad status code %v when getting UTXOs, retrying...\n", resp.StatusCode)
```

Comment:
Maybe sleep a couple secs and retry?

Reply:
This happens after waiting for a timeout from the server, which means some time has already passed and we should just retry immediately. 

The only reason not to do this would be if the timeout was random to offset lots of Darknodes retrying simultaneously. But, they’re not synchronised anyway so this shouldn’t be a major issue.